File: easy/calculate_distance.lua
for line in io.lines(arg[1]) do
    _, _, x1, y1, x2, y2 = string.find(line, "%((-?%d+), (-?%d+)%) %((-?%d+), (-?%d+)%)")
    local x = tonumber(x1)-tonumber(x2)
    local y = tonumber(y1)-tonumber(y2)
    print(math.floor(math.sqrt(x*x+y*y)))
end

File: easy/calculate_distance.hs
import System.Environment (getArgs)

stripChars :: String -> String -> String
stripChars = filter . flip notElem

distance   :: [Double] -> String
distance xs = show . floor $ sqrt (x*x + y*y)
            where x = head xs - xs!!2
                  y = xs!!1 - last xs

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (distance . map read . words) . lines $ stripChars "()," input
